ownCloudAppShared/Intent/SaveFileIntentHandler.swift

```diff
@@ -0,0 +1,151 @@
+//
+//  SaveFileIntentHandler.swift
+//  ownCloudAppShared
+//
+//  Created by Matthias Hühne on 30.07.19.
+//  Copyright © 2019 ownCloud GmbH. All rights reserved.
+//
+
+/*
+* Copyright (C) 2019, ownCloud GmbH.
+*
+* This code is covered by the GNU Public License Version 3.
+*
+* For distribution utilizing Apple mechanisms please see https://owncloud.org/contribute/iOS-license-exception/
+* You should have received a copy of this license along with this program. If not, see <http://www.gnu.org/licenses/gpl-3.0.en.html>.
+*
+*/
+
+import UIKit
+import Intents
+import ownCloudSDK
+
+@available(iOS 13.0, watchOS 6.0, *)
+public class SaveFileIntentHandler: NSObject, SaveFileIntentHandling {
+
+	var itemTracking : OCCoreItemTracking?
+
+	public func handle(intent: SaveFileIntent, completion: @escaping (SaveFileIntentResponse) -> Void) {
+		if AppLockHelper().isPassCodeEnabled {
+			completion(SaveFileIntentResponse(code: .authenticationRequired, userActivity: nil))
+		} else {
+			if let path = intent.path, let uuid = intent.account?.uuid, let file = intent.file, let fileURL = file.fileURL {
+				let accountBookmark = OCBookmarkManager.shared.bookmark(for: uuid)
+
+				if let bookmark = accountBookmark {
+					OCCoreManager.shared.requestCore(for: bookmark, setup: nil, completionHandler: { (core, error) in
+						if error == nil, let core = core {
+							self.itemTracking = core.trackItem(atPath: path, trackingHandler: { (error, item, isInitial) in
+								if let targetItem = item {
+									var newFilename = file.filename
+									if let filename = intent.filename as NSString?, filename.length > 0, let defaultFilename = file.filename as NSString? {
+										var pathExtention = defaultFilename.pathExtension
+										if let fileExtension = intent.fileextension, fileExtension.count > 0 {
+											pathExtention = fileExtension
+										}
+										if let changedFilename = filename.appendingPathExtension(pathExtention) {
+											newFilename = changedFilename
+										}
+									} else if let fileExtension = intent.fileextension, fileExtension.count > 0, let defaultFilename = file.filename as NSString? {
+										let filename = defaultFilename.deletingPathExtension as NSString
+										if let changedFilename = filename.appendingPathExtension(fileExtension) {
+											newFilename = changedFilename
+										}
+									}
+
+									if core.importFileNamed(newFilename,
+															at: targetItem,
```

Comment:
@hosy indentation issue? At least here it looks weird

Reply:
for me it looks ok, not sure what happened.